test/integration/auth/svcaccttoken_test.go

```diff
@@ -291,6 +319,166 @@ func TestServiceAccountTokenCreate(t *testing.T) {
+			}
+
+			info := doTokenReview(t, cs, treq, false)
+			if len(info.Extra) != len(expectedExtraValues) {
+				t.Fatalf("expected Extra have length of %d but was length %d: %#v", len(expectedExtraValues), len(info.Extra), info.Extra)
+			}
+			if !reflect.DeepEqual(info.Extra, expectedExtraValues) {
+				t.Fatalf("unexpected Extra:\ngot:\t%#v\nwant:\t%#v", info.Extra, expectedExtraValues)
+			}
+
+			delPod()
+			doTokenReview(t, cs, treq, true)
+		}
+	}
+
+	t.Run("bound to service account and a pod with an assigned nodeName that does not exist", testPodWithAssignedNode(nil))
+	t.Run("bound to service account and a pod with an assigned nodeName", testPodWithAssignedNode(node))
+
+	t.Run("bound to service account and node", func(t *testing.T) {
+		treq := &authenticationv1.TokenRequest{
+			Spec: authenticationv1.TokenRequestSpec{
+				Audiences: []string{"api"},
+				BoundObjectRef: &authenticationv1.BoundObjectReference{
+					Kind:       "Node",
+					APIVersion: "v1",
+					Name:       node.Name,
+					UID:        node.UID,
+				},
+			},
+		}
+
+		warningHandler.clear()
+		if resp, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{}); err == nil {
+			t.Fatalf("expected err creating token for nonexistant svcacct but got: %#v", resp)
+		}
+		warningHandler.assertEqual(t, nil)
+		sa, del := createDeleteSvcAcct(t, cs, sa)
+		defer del()
+
+		warningHandler.clear()
+		if resp, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{}); err == nil {
+			t.Fatalf("expected err creating token bound to nonexistant node but got: %#v", resp)
+		}
+		warningHandler.assertEqual(t, nil)
+		node, delNode := createDeleteNode(t, cs, node)
+		defer delNode()
+
+		// right uid
+		treq.Spec.BoundObjectRef.UID = node.UID
+		warningHandler.clear()
+		if _, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{}); err != nil {
+			t.Fatalf("err: %v", err)
+		}
+		warningHandler.assertEqual(t, nil)
+		// wrong uid
+		treq.Spec.BoundObjectRef.UID = wrongUID
+		warningHandler.clear()
+		if resp, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{}); err == nil {
+			t.Fatalf("expected err creating token bound to node with wrong uid but got: %#v", resp)
+		}
+		warningHandler.assertEqual(t, nil)
+		// no uid
+		treq.Spec.BoundObjectRef.UID = noUID
+		warningHandler.clear()
+		treq, err = cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+		warningHandler.assertEqual(t, nil)
+
+		checkPayload(t, treq.Status.Token, `"system:serviceaccount:myns:test-svcacct"`, "sub")
+		checkPayload(t, treq.Status.Token, `["api"]`, "aud")
+		checkPayload(t, treq.Status.Token, `null`, "kubernetes.io", "pod")
+		checkPayload(t, treq.Status.Token, `"test-node"`, "kubernetes.io", "node", "name")
```

Comment:
Also check the node UID is correct in the payload.

Reply:
done